tests/itest_specs/withdraw_liquidity.cairo

```diff
@@ -112,26 +109,192 @@ namespace WithdrawLiquidity {
+            strike_price=strike_price,
+            maturity=expiry,
+            option_side=0,
+            option_size=two,
+            quote_token_address=myusd_addr,
+            base_token_address=myeth_addr
+        );
+        
+        let (put_pool_unlocked_capital_1) = ILiquidityPool.get_unlocked_capital(
+            contract_address=amm_addr,
+            lptoken_address=lpt_put_addr
+        );
+        let (call_pool_unlocked_capital_1) = ILiquidityPool.get_unlocked_capital(
+            contract_address=amm_addr,
+            lptoken_address=lpt_call_addr
+        );
+
+        assert call_pool_unlocked_capital_1.low = 3012715374959942009;
+        assert put_pool_unlocked_capital_1.low = 2241738692;
+
+        let four_eth = Uint256(low = 5000000000000000000, high = 0);
+        ILiquidityPool.withdraw_liquidity(
+            contract_address=amm_addr,
+            pooled_token_addr=myeth_addr,
+            quote_token_address=myusd_addr,
+            base_token_address=myeth_addr,
+            option_type=0,
+            lp_token_amount=four_eth
+        );
+
+        let (call_pool_unlocked_capital_2) = ILiquidityPool.get_unlocked_capital(
+            contract_address=amm_addr,
+            lptoken_address=lpt_call_addr
+        );
+
+        // FIXME: this returns 340282366920938463461387345960256197662 
+        // assert call_pool_unlocked_capital_2.low = 3012715374959942009;
+
+        let four_thousand_usd = Uint256(low = 4000000000, high = 0);
+        ILiquidityPool.withdraw_liquidity(
+            contract_address=amm_addr,
+            pooled_token_addr=myusd_addr,
+            quote_token_address=myusd_addr,
+            base_token_address=myeth_addr,
+            option_type=1,
+            lp_token_amount=four_thousand_usd
+        );
+
+        let (put_pool_unlocked_capital_2) = ILiquidityPool.get_unlocked_capital(
+            contract_address=amm_addr,
+            lptoken_address=lpt_put_addr
+        );
+
+        // FIXME: this returns 340282366920938463463374607430011254584
+        // assert put_pool_unlocked_capital_2.low = 3012715374959942009;
+
+
+
         return ();
     }
 
-    func withdraw_liquidity_not_enough_lptokens{syscall_ptr: felt*, range_check_ptr}() {
-        // FIXME: TODO
-        // scenario when user is trying to withdraw more than he/she has
+    func withdraw_liquidity_not_enough_lptokens_call{syscall_ptr: felt*, range_check_ptr}() {
+        alloc_locals;
+
+        local amm_addr;
+        local myusd_addr;
+        local myeth_addr;
+        %{
+            ids.amm_addr = context.amm_addr
+            ids.myusd_addr = context.myusd_address
+            ids.myeth_addr = context.myeth_address
+
+            stop_prank_amm = start_prank(context.admin_address, context.amm_addr)
+
+            expect_revert(error_message = 'Failed to transfer token from pool to account in withdraw_liquidity')
+        %}
+
+        let six_eth = Uint256(low = 6000000000000000000, high = 0);
```

Comment:
withdraw is not in terms of ETH... it is in terms of LP tokens... so the lp_token_amount is how much lp tokens you want to burn and the number of ETH you received is calculated based on the value of those burned LP tokens